src/lib/repositories/impl_oauth2/oauth2_repository_impl.py

```diff
@@ -0,0 +1,195 @@
+from injector import inject
+from sqlalchemy import text
+from sqlalchemy.engine.base import Engine
+import jwt
+from datetime import datetime, timedelta
+
+from src.constants.http_status_code import HttpStatus
+from src.env_config import get_env_config_instance
+from src.utils.sql_oath_queries import (
+    SQL_QUERY_TO_GET_USER_BY_USER_NAME,
+    SQL_QUERY_TO_GET_CLIENT_BY_CLIENT_ID,
+    SQL_QUERY_TO_GET_CLIENT_BY_ID,
+    SQL_QUERY_TO_ADD_ACCESS_TOKEN,
+    SQL_QUERY_TO_ADD_REFRESH_TOKEN,
+    SQL_QUERY_TO_GET_REFRESH_TOKEN_BY_USER_NAME,
+    SQL_QUERY_TO_GET_ACCESS_TOKEN_BY_TOKEN,
+    SQL_QUERY_TO_GET_REFRESH_TOKEN_BY_ID,
+    SQL_QUERY_TO_DELETE_ACCESS_TOKEN,
+    SQL_QUERY_TO_DELETE_REFRESH_TOKEN,
+)
+
+
+class Oauth2Services:
+    @inject
+    def __init__(self, engine: Engine):
+        self.env_config = get_env_config_instance()
+        self.engine = engine
+
+    def login(self, user_name, password, client_id):
```

Comment:
I don't see the client secret here

Reply:
There should be 2 logins, one login only for that accepts user credentials + client credentials and another one only for client credentials. Also there should be one login accepting all of the above + credentials type

Reply:
done.